client/src/components/pages/Stats.js

```diff
@@ -0,0 +1,314 @@
+import React, { useState, useEffect } from "react";
+import "../../utilities.css";
+import "./Stats.css";
+import { get, post, prettifyTourney, hasAccess, getStage } from "../../utilities";
+import { Layout, Table, Menu, Form, Switch, message } from "antd";
+const { Content } = Layout;
+const { Column, ColumnGroup } = Table;
+import StageSelector from "../modules/StageSelector";
+
+export default function Stats({ tourney, user }) {
+  const [state, setState] = useState({});
+
+  const fetchData = async () => {
+    const [tourneyModel, currentSelectedStage] = await getStage(tourney);
+
+    const players = await get("/api/players", { tourney });
+
+    let stageMaps = [];
+    try {
+      stageMaps = await get("/api/maps", { tourney, stage: currentSelectedStage.name });
+    } catch {}
+
+    let stageStats = {};
+    try {
+      stageStats = await get("/api/stage-stats", { tourney, stage: currentSelectedStage.name });
+    } catch {}
```

Comment:
could optimize performance a bit by firing off those api calls at the same time instead of awaiting them sequentially

```js
const [players, stageMaps, stageStats] = await Promise.all([
  get("/api/players", { tourney }),
  get("/api/maps", { tourney, stage: currentSelectedStage.name }),
  get("/api/stage-stats", { tourney, stage: currentSelectedStage.name }),
]);
```

Reply:
I wanted to try catch the `maps` and `stage-stats` calls since they can return errors, so the rest of the page can load, instead of showing somewhat broken page:
![Screenshot 2022-03-15 16 22 05](https://user-images.githubusercontent.com/4967258/158465296-4387f030-57ab-463c-bfaa-9951c6c91c7c.png)

Reply:
ahh okay nice